org.sofproject.alsa.topo/src/org/sofproject/alsa/topo/model/AlsaTopoGraph.java

```diff
@@ -536,7 +515,35 @@ public void serialize() throws CoreException, IOException {
 		}
 		outputFile.getParent().refreshLocal(1, null);
 	}
-
+	
+	@Override
+	public String getPipelineString() {
+		try {
+			ByteArrayOutputStream os = new ByteArrayOutputStream();
+			Writer writer = new BufferedWriter(new OutputStreamWriter(os));
+			tlvs.serialize(writer);
```

Comment:
Do not duplicate the code of ITopoGraph.serialize(). Atm me may just refactor the latter to return a String and move target file content writing to topo.ui.